### DISCUSSION TITLE
How to reduce the resulting wasm file size?

### DISCUSSION THREAD

**Discussion by AfflatusX** [wasmerio/wasmer#3822] [General]
### Summary

Are there general guidance on how to reduce the size of the .wasm file produced by wasicc? 

I tried to use gcc to compile the code natively which yields a 700k binary without -Wl,-dead_strip flags on macOS. With those options the binary is about 250k.

Using wasicc the binary is about 1.4mb - but not sure what compiler options can I use to trim the size down.

Any guidance on how to optimize for a smaller binary?

**Comment by mucinoab** [comment]
I found this recommendations. 
[Bevy - WASM, optimizing](https://github.com/bevyengine/bevy/tree/main/examples#2-use-wasm-opt-from-the-binaryen-package)

**Comment by ptitSeb** [comment]
Moving that to discussion
